libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -614,6 +630,40 @@ - (void)sendRequestWithSmartSyncUserAgent:(SFRestRequest *)request failBlock:(SF
     [self.restClient sendRESTRequest:request failBlock:failBlock completeBlock:completeBlock];
 }
 
+- (BOOL)isNewerThanServer:(NSString *)objectType objectId:(NSString *)objectId lastModifiedDate:(long long)lastModifiedDate {
+    __block BOOL isNewer = NO;
+    __block long long serverLastModified = -1;
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+    SFSmartSyncSoqlBuilder *soqlBuilder = [SFSmartSyncSoqlBuilder withFields:kSyncManagerLastModifiedDate];
+    [soqlBuilder from:objectType];
+    [soqlBuilder where:[NSString stringWithFormat:@"Id = '%@'", objectId]];
+    NSString *query = [soqlBuilder build];
+    SFRestRequest *request = [[SFRestAPI sharedInstance] requestForQuery:query];
+    [self sendRequestWithSmartSyncUserAgent:request
+        failBlock:^(NSError *error) {
+            [self log:SFLogLevelError format:@"REST request failed with error: %@", error];
+            dispatch_semaphore_signal(semaphore);
+        }
+        completeBlock:^(NSDictionary* d) {
+            if (nil != d) {
+                NSDictionary *record = d[@"records"][0];
+                if (nil != record) {
+                    NSString *serverLastMod = record[kSyncManagerLastModifiedDate];
+                    if (nil != serverLastMod) {
+                        serverLastModified = [self getTimeMillisFromString:serverLastMod];
+                    }
+                }
+            }
+            if (serverLastModified <= lastModifiedDate) {
+                isNewer = YES;
+            }
+            dispatch_semaphore_signal(semaphore);
+        }
+     ];
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
```

Comment:
In that case, we could rewrite the whole sync up to be more readable (and a simple loop when calling syncUpOneEntry)